src/generation.rs

```diff
@@ -1278,8 +1278,7 @@ impl GenerationScope {
                         compare_block.line(format!("return Err(DeserializeFailure::FixedValueMismatch{{ found: Key::Float({}_value), expected: Key::Float({}) }}.into());", config.var_name, x));
                         deser_code.content.push_block(compare_block);
                         if CLI_ARGS.preserve_encodings {
-                            config.final_exprs.push(format!("Some({}_encoding)", config.var_name));
-                            deser_code.content.line(&format!("{}{}{}", before_after.before_str(false), final_expr(config.final_exprs, None), before_after.after_str(false)));
+                            unimplemented!("preserve_encodings is not implemented for float")
```

Comment:
This would only be hit for constants, wouldn't it? We should error for when a user has non-const floats too i.e. in the `SerializingRustType::Root(ConceptualRustType::Primitive(p))` case below. Same for when floats are used as keys. There might be some other area too, but off the top of my head there are those two. Or we could put the error somewhere more general that would be hit by all such floats if possible.

Reply:
About which keys are you talking? Keys in a map are failing with `thread 'main' panicked at 'unsupported map key type for MyMap.key_3.3: Float(3.3)', src\generation.rs:3129:26`

Reply:
Ah sorry, I didn't realize that this PR didn't add float keys. That's fine then as it's caught by that previous panic.